sdk/cosmosdb/cosmos/review/cosmos.api.md

```diff
@@ -90,83 +93,86 @@ export class ChangeFeedResponse<T> {
+    bulk<T>({ body, path, partitionKeyRangeId, resourceId, bulkOptions, options, diagnosticNode, }: {
         body: T;
         path: string;
         partitionKeyRangeId: string;
         resourceId: string;
         bulkOptions?: BulkOptions;
         options?: RequestOptions;
-        diagnosticContext?: CosmosDiagnosticContext;
+        diagnosticNode: DiagnosticNodeInternal;
     }): Promise<Response_2<any>>;
     // (undocumented)
     clearSessionToken(path: string): void;
     // (undocumented)
-    create<T, U = T>({ body, path, resourceType, resourceId, options, partitionKey, diagnosticContext, }: {
+    create<T, U = T>({ body, path, resourceType, resourceId, diagnosticNode, options, partitionKey, }: {
         body: T;
         path: string;
         resourceType: ResourceType;
         resourceId: string;
+        diagnosticNode: DiagnosticNodeInternal;
         options?: RequestOptions;
         partitionKey?: PartitionKey;
-        diagnosticContext?: CosmosDiagnosticContext;
     }): Promise<Response_2<T & U & Resource>>;
     // (undocumented)
-    delete<T>({ path, resourceType, resourceId, options, partitionKey, method, diagnosticContext, }: {
+    delete<T>({ path, resourceType, resourceId, options, partitionKey, method, diagnosticNode, }: {
         path: string;
         resourceType: ResourceType;
         resourceId: string;
         options?: RequestOptions;
         partitionKey?: PartitionKey;
         method?: HTTPMethod;
-        diagnosticContext?: CosmosDiagnosticContext;
+        diagnosticNode: DiagnosticNodeInternal;
     }): Promise<Response_2<T & Resource>>;
     // (undocumented)
-    execute<T>({ sprocLink, params, options, partitionKey, diagnosticContext, }: {
+    execute<T>({ sprocLink, params, options, partitionKey, diagnosticNode, }: {
         sprocLink: string;
         params?: any[];
         options?: RequestOptions;
         partitionKey?: PartitionKey;
-        diagnosticContext?: CosmosDiagnosticContext;
+        diagnosticNode: DiagnosticNodeInternal;
     }): Promise<Response_2<T>>;
-    getDatabaseAccount(options?: RequestOptions, diagnosticContext?: CosmosDiagnosticContext): Promise<Response_2<DatabaseAccount>>;
     // (undocumented)
-    getQueryPlan(path: string, resourceType: ResourceType, resourceId: string, query: SqlQuerySpec | string, options?: FeedOptions, diagnosticContext?: CosmosDiagnosticContext): Promise<Response_2<PartitionedQueryExecutionInfo>>;
+    getClientConfig(): ClientConfig;
+    getDatabaseAccount(diagnosticNode: DiagnosticNodeInternal, options?: RequestOptions): Promise<Response_2<DatabaseAccount>>;
     // (undocumented)
-    getReadEndpoint(): Promise<string>;
+    getQueryPlan(path: string, resourceType: ResourceType, resourceId: string, query: SqlQuerySpec | string, options: FeedOptions, diagnosticNode: DiagnosticNodeInternal): Promise<Response_2<PartitionedQueryExecutionInfo>>;
```

Comment:
options argument type changed to required. Any specific reason for that?

Reply:
`ClientContext` is an internal method. `api-extractor` was marking `options` argument as optional because it was initialized as - 
public async getQueryPlan(
    path: string,
    resourceType: ResourceType,
    resourceId: string,
    query: SqlQuerySpec | string,
    options: FeedOptions = {}
}
Notice the default value of {}. Now that `diagnosticNode: DiagnosticNodeInternal` parameter is added after this, it is marked as non-optional. Which is harmless.